portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -1,24 +1,37 @@
 package com.google.sps.servlets;
 
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 @WebServlet("/form-handler")
 public class FormHandlerServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    // Get the value entered in the form.
-        String textValue = request.getParameter("text-input");
+    String name = request.getParameter("name");
+    String email = request.getParameter("email") ;
+    String message = request.getParameter("message");
 
-    // Print the value so you can see it in the server logs.
-        System.out.println("You submitted: " + textValue);
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory().setKind("Task");
+    FullEntity taskEntity =
+        Entity.newBuilder(keyFactory.newKey())
+            .set("name", name)
+            .set("email", email)
```

Comment:
So, this is kind of glossed over in the walkthroughs, but you generally can't collect and store a user's personal data like name or email without having extra security around the data and explicit consent to collect the data from the user. 

How to properly secure the data and what privacy notices must be given are probably beyond the scope of SPS because there's a lot of detail that goes into it, and it varies from region to region.

For the sake of the datastore tutorial, I'd recommend only collecting a "username" which doesn't imply real name, and dropping "name" and "email".

Reply:
Ok, asked some other advisors about storing the email, and the consensus is it's ok to collect an email for personal or "household" applications. Essentially non-commercial applications, and this definitely is not a commercial application.

At Google, we'd have to jump through a few more hoops to collect an e-mail, but you are good to go for your personal portfolio.

It's good to keep user privacy and securing their data in mind though when collecting personal information. At most companies there's legal teams to help you out with this because, as I have demonstrated, it is hard for us developers to keep track of it all.